builtin/logical/database/backend.go

```diff
@@ -9,13 +9,37 @@ import (
 
 	log "github.com/mgutz/logxi/v1"
 
+	"github.com/hashicorp/errwrap"
+	uuid "github.com/hashicorp/go-uuid"
 	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
 	"github.com/hashicorp/vault/logical"
 	"github.com/hashicorp/vault/logical/framework"
+	"github.com/hashicorp/vault/plugins/helper/database/dbutil"
 )
 
 const databaseConfigPath = "database/config/"
 
+type dbPluginInstance struct {
```

Comment:
Took me a bit to figure this out, but IIUC `dbPluginInstance` is a _connection_ instance of a particular database plugin (I took its as an instance of the plugin process on first read). I think it'd be helpful to rename it to `dbConnectionInstance` or something along those lines.

Reply:
It's really a wrapper around the database.Plugin interface which in this context is the client.  I named it this way since it is the instance of the original object that it was wrapping.